### ISSUE TITLE
模型做imputation内部标准化问题

### ISSUE THREAD

**Issue by mbhnwz** [kwuking/TimeMixer#128]
你好，kwuking
最近我阅读到您TimeMixer与TimeMixer++的论文，觉得你的工作很有趣，并进一步想要阅读其源码。在阅读源码的过程中，我遇到了一些问题想要向您咨询，具体描述为：
1、在模型做imputation工作时，模型内部基于每个特征进行了标准化处理，请问在数据加载时已经标准化，然而在模型内部进行这种新的标准化，这样做的目的是什么。（代码如下文所示）
2、当存在某个样本某个特征数据全为未知（mask均为0）时，means值计算会出现0除nan值，请问这种情况是否需要特殊处理？
means = torch.sum(x_enc, dim=1) / torch.sum(mask == 1, dim=1)
means = means.unsqueeze(1).detach()
x_enc = x_enc - means
x_enc = x_enc.masked_fill(mask == 0, 0)
stdev = torch.sqrt(torch.sum(x_enc * x_enc, dim=1) /
                   torch.sum(mask == 1, dim=1) + 1e-5)
stdev = stdev.unsqueeze(1).detach()
x_enc /= stdev

期待您的解答，多谢！

**Comment by kevinliu2000** [comment]
您好，非常感谢您对我们工作的支持，这里我们是参考自TSLib中的做法，这个想法我理解是来自于non-stationary transformers这篇论文提出的观点，对数据在归一化之外去在输入的context序列上去做平稳化处理；关于您提到的第二个问题，我个人理解如果是0值较多的数据，确实是需要进行padding的平滑处理，目前代码中已经支持对于这种情况的处理。
